inc/fields/class-field-user-select.php

```diff
@@ -0,0 +1,174 @@
+			self::$instance->setup_actions();
+		}
+		return self::$instance;
+	}
+
+	/**
+	 * Add the required actions and filters.
+	 */
+	private function setup_actions() {
+
+		add_filter( 'shortcode_ui_fields',             array( $this, 'filter_shortcode_ui_fields' ) );
+		add_action( 'enqueue_shortcode_ui',            array( $this, 'action_enqueue_shortcode_ui' ) );
+		add_action( 'wp_ajax_shortcode_ui_user_field', array( $this, 'action_wp_ajax_shortcode_ui_user_field' ) );
+		add_action( 'shortcode_ui_loaded_editor',      array( $this, 'action_shortcode_ui_loaded_editor' ) );
+	}
+
+	/**
+	 * Add our field to the shortcode fields.
+	 *
+	 * @param $fields
+	 *
+	 * @return array
+	 */
+	public function filter_shortcode_ui_fields( $fields ) {
+
+		return array_merge( $fields, $this->fields );
+	}
+
+	/**
+	 * Add Select2 for our UI.
+	 */
+	public function action_enqueue_shortcode_ui() {
+
+		$plugin_dir = dirname( dirname( __FILE__ ) );
+
+		wp_enqueue_script( 'select2', plugins_url( 'lib/select2/select2.min.js', $plugin_dir ), array( 'jquery', 'jquery-ui-sortable' ), '3.5.2' );
+		wp_enqueue_style( 'select2', plugins_url( 'lib/select2/select2.css', $plugin_dir ), null, '3.5.2' );
+
+		wp_localize_script( 'shortcode-ui', 'shortcodeUiUserFieldData', array(
+			'nonce' => wp_create_nonce( 'shortcode_ui_field_user_select' ),
+		) );
+	}
+
+	/**
+	 * Output styles and templates used by user select field.
+	 */
+	public function action_shortcode_ui_loaded_editor() {
+		?>
+
+		<style>
+
+			.edit-shortcode-form .select2-container {
+				min-width: 300px;
+			}
+
+			.edit-shortcode-form .select2-container a {
+				transition: none;
+				-webkit-transition: none;
+			}
+
+			.wp-admin .select2-drop {
+				z-index: 160001;
+			}
+
+		</style>
+
+		<script type="text/html" id="tmpl-shortcode-ui-field-user-select">
+			<div class="field-block shortcode-ui-field-user-select shortcode-ui-attribute-{{ data.attr }}">
+				<label for="{{ data.id }}">{{{ data.label }}}</label>
+				<input type="text" name="{{ data.attr }}" id="{{ data.id }}" value="{{ data.value }}" class="shortcode-ui-user-select" />
+				<# if ( typeof data.description == 'string' ) { #>
+					<p class="description">{{{ data.description }}}</p>
+				<# } #>
+			</div>
+		</script>
+
+		<?php
+	}
+
+	/**
+	 * Ajax handler for select2 user field queries.
+	 * Output JSON containing user data.
+	 * Requires that shortcode, attr and nonce are passed.
+	 * Requires that the field has been correctly registered and can be found in $this->post_fields
+	 * Supports passing page number and search query string.
+	 *
+	 * @return null
+	 */
+	public function action_wp_ajax_shortcode_ui_user_field() {
+
+		$nonce               = isset( $_GET['nonce'] ) ? sanitize_text_field( $_GET['nonce'] ) : null;
+		$requested_shortcode = isset( $_GET['shortcode'] ) ? sanitize_text_field( $_GET['shortcode'] ) : null;
+		$requested_attr      = isset( $_GET['attr'] ) ? sanitize_text_field( $_GET['attr'] ) : null;
```

Comment:
@dashaluna sure, WordPress has an unfortunately legacy back-compat feature where it will force all of the input vars like `$_GET` and `$_POST` to be “slashed”. So for example, if you put in a search for “I'm happy”, then if you look at `$_GET['s']` it will actually be `I\'m happy`. 

For more information, see https://core.trac.wordpress.org/ticket/18322